EIPS/eip-5773.md

```diff
@@ -0,0 +1,362 @@
+    function getResourceMetaForToken(uint256 tokenId, uint64 resourceIndex) external view returns (string memory);
+
+    /// @notice Returns the IDs of all of the stored resources.
+    function getAllResources() external view returns (uint64[] memory);
+
+    /// @notice Change or reaffirm the approved address for resources for an .
+    /// @dev The zero address indicates there is no approved address.
+    /// @dev MUST revert unless `msg.sender` is the current NFT owner, or an authorized
+    ///  operator of the current owner.
+    /// @dev MUST emit ApprovalForResources event.
+    /// @param to The new approved token controller
+    /// @param tokenId The ID of the token to approve
+    function approveForResources(address to, uint256 tokenId) external payable;
+
+    /// @notice Enable or disable approval for a third party ("operator") to manage
+    ///  resources for all of the `msg.sender`'s assets.
+    /// @dev MUST emit the ApprovalForAllForResources event.
+    /// @dev The contract MUST allow multiple operators per owner.
+    /// @param operator Address to add to the set of authorized operators
+    /// @param approved True if the operator is approved, false to revoke approval
+    function setApprovalForAllForResources(address operator, bool approved) external;
+
+    /// @notice Get the approved address for resources for a single token.
+    /// @dev MUST revert if `tokenId` is not a valid token ID.
+    /// @param tokenId The ID of the token to find the approved address for
+    /// @return The approved address for this token, or the zero address if there is none
+    function getApprovedForResources(uint256 tokenId) external view returns (address);
+
+    /// @notice Query if an address is an authorized operator for resources of
+    ///  another address.
+    /// @param owner The address that owns the tokens
+    /// @param operator The address that acts on behalf of the owner
+    /// @return True if `operator` is an approved operator for `owner`, false otherwise
+    function isApprovedForAllForResources(address owner, address operator) external view returns (bool);
+
+}
+
+interface ERC165 {
+
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+
+}
+````
+
+## Rationale
+
+With NFTs being a widespread form of tokens in the Ethereum ecosystem and being used for a variety of use cases, it is time to standardize additional utility for them. Having multiple resources associated with a single NFT allows for greater utility and usability.
+
+The cross-metaverse or rather cross-engine compatibility issue could be solved by this EIP by having resources compatible with multiple engines or metaverses associated with the same NFT.
+
+Such NFT can be given further utility by means of new additional resources: more games, more cosmetic items, appended to the same NFT. Thus, a game cosmetic item as an NFT becomes an ever-evolving NFT of infinite utility.
```

Comment:
This is all Motivation, not Rationale. The Rationale section should explain individual technical choices made within this EIP, while the Motivation should justify the existence of the EIP as a whole.